Searcher: factor out structural search logic

jtibshirani opened the pull request:
Searcher's entrypoint method handles delegating to standard unindexed search
vs. structural search. This change consolidates the structural search logic to
make this method easier to follow

Specific changes:
* Create new function `Service.getZipFile`
* Consolidate structural search logic
* Remove redundant method `Store.PrepareZip`

## Test plan

Refactor is covered by existing unit tests

camdencheek reviewed (approved):
much easier to follow!

> Remove redundant method Store.PrepareZip

What do you mean by this? Looks like `PrepareZip` is still around

jtibshirani commented:
Thanks for the speedy review.

> What do you mean by this? Looks like PrepareZip is still around

I removed the old `PrepareZip` and renamed `PrepareZipPaths` -> `PrepareZip`
